src/RatcliffDDM.jl

```diff
@@ -262,4 +262,187 @@ logpdf(d::RatcliffDDM, data::Tuple) = logpdf(d, data...)
+        ν = sign(ν) * 5
+        warn("ν is not in the range [-5, 5], bounding drift rate to $Nu...")
+    end
+
+    if η > 3
+        warn("Standard deviation of drift rate is out of bounds, bounding drift rate to 3")
+        η = 3
+    end
+
+    if η == 0
+        η = 1e-16
+    end
+
+    # Initialize output vectors
+    result = zeros(N)
+    T = zeros(N)
+    XX = zeros(N)
+
+    # Called sigma in 2001 paper
+    D = σ^2 / 2
+
+    # Program specifications
+    ϵ = 2.220446049250313e-16  # precision from 1.0 to next double-precision number
+    Δ = ϵ
+
+    for n in 1:N
+        r1 = randn()
+        μ = ν + r1 * η
+        bb = z - sz / 2 + sz * rand()
+        zz = bb * α
+        finish = 0
+        totaltime = 0
+        startpos = 0
+        Aupper = α - zz
+        Alower = -zz
+        radius = min(abs(Aupper), abs(Alower))
+
+        while finish == 0
+            λ = 0.25 * μ^2 / D + 0.25 * D * π^2 / radius^2
+            # eq. formula (13) in 2001 paper with D = sigma^2/2 and radius = Alpha/2
+            F = D * π / (radius * μ)
+            F = F^2 / (1 + F^2)
+            # formula p447 in 2001 paper
+            prob = exp(radius * μ / D)
+            prob = prob / (1 + prob)
+            dir_ = 2 * (rand() < prob) - 1
+            l = -1
+            s2 = 0
+            s1 = 0
+            while s2 > l
+                s2 = rand()
+                s1 = rand()
+                tnew = 0
+                told = 0
+                uu = 0
+                while abs(tnew - told) > ϵ || uu == 0
+                    told = tnew
+                    uu += 1
+                    tnew = told + (2 * uu + 1) * (-1)^uu * s1^(F * (2 * uu + 1)^2)
+                    # infinite sum in formula (16) in BRMIC,2001
+                end
+                l = 1 + s1^(-F) * tnew
+            end
+            # rest of formula (16)
+            t = abs(log(s1)) / λ
+            # is the negative of t* in (14) in BRMIC,2001
+            totaltime += t
+            dir_ = startpos + dir_ * radius
+            ndt = τ - st / 2 + st * rand()
+            if (dir_ + Δ) > Aupper
+                T[n] = ndt + totaltime
+                XX[n] = 1
+                finish = 1
+            elseif (dir_ - Δ) < Alower
+                T[n] = ndt + totaltime
+                XX[n] = 2
+                finish = 1
+            else
+                startpos = dir_
+                radius = minimum(abs.([Aupper, Alower] .- startpos))
+            end
+        end
+    end
+    return (choice=XX,rt=T)
+end
+
+function _rand_stochastic(rng::AbstractRNG, d::RatcliffDDM; N::Int = 1, nsteps::Int=300, step_length::Int=0.01)
+    (ν, α, τ, z, η, sz, st, σ) = params(d)
+
+    if (ν < -5) || (ν > 5)
+        ν = sign(ν) * 5
+        warn("ν is not in the range [-5, 5], bounding drift rate to $Nu...")
+    end
+
+    if η > 3
+        warn("Standard deviation of drift rate is out of bounds, bounding drift rate to 3")
+        η = 3
+    end
+
+    if η == 0
```

Comment:
I like the idea of suggesting typical parameter ranges in the documentation, but I don't think we should enforce bounds unless they are strict (e.g., a standard deviation must be non-negative). I think the best approach is to allow users to enforce their own parameter bounds. This will also be consistent with the approach taken for the other models in the package. .

Reply:
Okay. I will make a separate PR that contains an implementation for this across models.